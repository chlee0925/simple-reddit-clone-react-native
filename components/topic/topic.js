import React from 'react';
import { View, Text, StyleSheet, Button } from 'react-native';

/**
 * Component that renders a topic. User can vote up or vote down on the topic.
 */
export class Topic extends React.Component {
    render() {
        return (
            <View style={style.container}>
                <Text style={style.topic}>Topic: {this.props.topic}</Text>
                <View style={style.propertyContainer}>
                    <Text>index: {this.props.index}</Text>
                    <Text>upvote: {this.props.upvote}</Text>
                    <Text>downvote: {this.props.downvote}</Text>
                </View>
                
                <View style={style.buttonContainer}>
                    <Button
                        title='Vote Up'
                        onPress={() => this.props.upvoteFn(this.props.index)}
                     />
                     <Text style={style.voteSum}>Vote Sum: {this.props.upvote - this.props.downvote}</Text>
                     <Button
                        title='Vote Down'
                        onPress={() => this.props.downvoteFn(this.props.index)}
                     />
                </View>
            </View>
        )
    }
}

const style = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#f2f2f2',
        padding: 5,
        marginBottom: 5,
        borderTopWidth: 1,
        borderBottomWidth: 2,
        borderColor: '#cccccc'
    },
    topic: {
        fontSize: 18,
        fontWeight: 'bold',
        marginBottom: 10
    },
    propertyContainer: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'space-around',
        marginBottom: 10
    },
    buttonContainer: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center'
    },
    voteSum: {
        fontSize: 13,
        fontWeight: 'bold'
    }
});
