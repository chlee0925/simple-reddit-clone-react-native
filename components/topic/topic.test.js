import React from 'react';
import { Topic } from './topic.js';

import renderer from 'react-test-renderer';

it('renders Topic Component', () => {
    const mockTopicProps = {
        index: 1,
        item: {
            topic: 'test topic',
            upvote: 1,
            downvote: 2,
            upvoteFn: jest.fn(),
            downvoteFn: jest.fn()
        }
    }
    const screen = renderer.create(
        <Topic
            index={mockTopicProps.index}
            topic={mockTopicProps.item.topic}
            upvote={mockTopicProps.item.upvote}
            downvote={mockTopicProps.item.downvote}
            upvoteFn={mockTopicProps.item.upvoteFn}
            downvoteFn={mockTopicProps.item.downvoteFn}
        />
    ).toJSON();
    expect(screen).toMatchSnapshot();
});