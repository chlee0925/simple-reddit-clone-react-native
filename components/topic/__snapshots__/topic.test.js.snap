// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Topic Component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#e3b5bc",
      "flex": 1,
      "marginBottom": 3,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 10,
      }
    }
  >
    Topic: 
    test topic
  </Text>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
        "marginBottom": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      index: 
      1
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      upvote: 
      1
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      downvote: 
      2
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Vote Up
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Vote Down
        </Text>
      </View>
    </View>
  </View>
</View>
`;
